pandas/tests/series/indexing/test_setitem.py

```diff
@@ -527,14 +529,17 @@ def test_setitem_empty_series_timestamp_preserves_dtype(self):
             Timedelta("9 days").to_pytimedelta(),
         ],
     )
-    def test_append_timedelta_does_not_cast(self, td):
+    def test_append_timedelta_does_not_cast(self, td, using_infer_string):
         # GH#22717 inserting a Timedelta should _not_ cast to int64
         expected = Series(["x", td], index=[0, "td"], dtype=object)
 
         ser = Series(["x"])
         ser["td"] = td
         tm.assert_series_equal(ser, expected)
-        assert isinstance(ser["td"], Timedelta)
+        if using_infer_string and not isinstance(td, Timedelta):
+            assert not isinstance(ser["td"], Timedelta)
```

Comment:
What would `ser["td"]` be here?

Reply:
one of those 2:

```
            Timedelta("9 days").to_timedelta64(),
            Timedelta("9 days").to_pytimedelta(),
```

maybe we should xfail that test, not sure anymore what my reasoning here was

Reply:
Yeah I find it weird to begin with that this always promoted timedeltas to `pd.Timedelta`.

Agree to xfail this test for now to nail down the correct behavior in a follow up